src/Shared/RunnerUtilities.cs

```diff
@@ -0,0 +1,102 @@
+﻿using Microsoft.Build.Shared;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Build.SharedUtilities
+{
+
+    internal static class RunnerUtilities
+    {
+        public static string PathToMsBuildExe => FileUtilities.CurrentExecutablePath;
+
+        /// <summary>
+        /// Invoke msbuild.exe with the given parameters and return the stdout, stderr, and process exit status.
+        /// This method may invoke msbuild via other runtimes.
+        /// </summary>
+        public static string ExecMSBuild(string msbuildParameters, out bool successfulExit)
+        {
+            return ExecMSBuild(PathToMsBuildExe, msbuildParameters, out successfulExit);
+        }
+
+        /// <summary>
+        /// Invoke msbuild.exe with the given parameters and return the stdout, stderr, and process exit status.
+        /// This method may invoke msbuild via other runtimes.
+        /// </summary>
+        public static string ExecMSBuild(string pathToMsBuildExe, string msbuildParameters, out bool successfulExit)
+        {
+#if FEATURE_RUN_EXE_IN_TESTS
```

Comment:
I did not know whether this functionality would be useful for non-test code too so I treated it the same as FileUtilities, since there should be a single place that knows how to resolve the path to msbuild.exe and to invoke it as well. 

Should I move it to a shared test folder instead?